### PR TITLE
Update API Gateway controller to new codegen structure

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

In order to get the latest features, we need to update all ACK-generated controllers to new hook functions.

The first commit is only generation, the second one is manually done and that's the part that needs to be reviewed.

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested `API` resource just to be sure but didn't go through all of them since there isn't much functionality change.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/controller/apigatewayv2/api/setup.go [hunk 1]
```diff
@@ -36,30 +36,36 @@ import (
...
+func preObserve(_ context.Context, cr *svcapitypes.API, obj *svcsdk.GetApiInput) error {
+	obj.ApiId = aws.String(meta.GetExternalName(cr))
 	return nil
 }
-func (*external) postObserve(_ context.Context, cr *svcapitypes.API, _ *svcsdk.GetApiOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+
+func postObserve(_ context.Context, cr *svcapitypes.API, _ *svcsdk.GetApiOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
 	cr.SetConditions(xpv1.Available())
 	return obs, err
 }
```
Comment (hasheddan) [inline pkg/controller/apigatewayv2/api/setup.go:66]: Do we need to check error before setting `Available` here?

Comment (hasheddan) [reply pkg/controller/apigatewayv2/api/setup.go:66]: It does seem like in practice we are always passing `nil` error here, but I see it is checked in other places, so not sure what pattern we want to follow

**Review by hasheddan — COMMENTED** [review]
Looks pretty good to me @muvaf! Awesome work, per usual. I did leave one comment. Overall, I like the model of generating the no-op's then overriding with options rather than having to hand write all of the methods that just return the `obj` without any operations. Have you had the opportunity to do manual testing here?

File: pkg/controller/apigatewayv2/api/setup.go (hunk 1 above)
Comment (muvaf) [reply pkg/controller/apigatewayv2/api/setup.go]: Yeah, good catch!

**Comment by muvaf** [comment]
@hasheddan I just updated the testing section, I tested the `API` resource for sanity check but not all of them.

**Review by hasheddan — APPROVED** [review]

**Pull request merged**